backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/VersionInputHandler.java

```diff
@@ -0,0 +1,245 @@
+                  .bind("commit", versioningEntry.getCommit())
+                  .bind("versioning_key", locationEntry.getKey())
+                  .bind(
+                      "versioning_location",
+                      ModelDBUtils.getStringFromProtoObject(locationEntry.getValue()))
+                  .bind("versioning_blob_type", blob.getContentCase().getNumber())
+                  .bind("blob_hash", blobExpandedWithHashMap.getValue())
+                  .bind("entity_type", entity_type)
+                  .bind("entityId", entityId)
+                  .execute();
+
+              if (blob.getContentCase().equals(Blob.ContentCase.CONFIG)) {
+                var configBlobQuery =
+                    "SELECT cb.blob_hash, cb.config_seq_number, cb.hyperparameter_type, hecbs.name, hecbs.int_value, hecbs.float_value, hecbs.string_value "
+                        + " FROM config_blob cb "
+                        + " JOIN hyperparameter_element_config_blob hecbs ON cb.hyperparameter_element_config_blob_hash = hecbs.blob_hash"
+                        + " WHERE cb.blob_hash = :blobHash";
+                handle
+                    .createQuery(configBlobQuery)
+                    .bind("blobHash", blobExpandedWithHashMap.getValue())
+                    .map(
+                        (rs, ctx) -> {
+                          String blobMappingQueryStr =
+                              " INSERT INTO versioning_modeldb_entity_mapping_config_blob "
+                                  + " (versioning_modeldb_entity_mapping_repository_id, "
+                                  + " versioning_modeldb_entity_mapping_commit, "
+                                  + " versioning_modeldb_entity_mapping_versioning_key, "
+                                  + " versioning_modeldb_entity_mapping_"
+                                  + entityIdReferenceColumn
+                                  + ", "
+                                  + " versioning_modeldb_entity_mapping_entity_type, "
+                                  + " config_blob_entity_blob_hash, "
+                                  + " config_blob_entity_config_seq_number) "
+                                  + " VALUES (:repository_id, :commit, :versioning_key, :entityId, :entity_type, :blob_hash, :config_seq_number)";
+                          handle
+                              .createUpdate(blobMappingQueryStr)
+                              .bind("repository_id", versioningEntry.getRepositoryId())
+                              .bind("commit", versioningEntry.getCommit())
+                              .bind("versioning_key", locationEntry.getKey())
+                              .bind("entityId", entityId)
+                              .bind("entity_type", entity_type)
+                              .bind("blob_hash", rs.getString("blob_hash"))
+                              .bind("config_seq_number", rs.getString("config_seq_number"))
+                              .execute();
```

Comment:
Don't insert in the middle of mapping a select.

Reply:
We are selecting mappings to insert this data by use of fetch one so this is a correct place i think so..

Reply:
It is not. Don't do a DB operation while in the middle of another operation. That is really bad practice that kills performance. Do one thing at a time.

Reply:
separated in multiple functions instead of middle of DB operation